### PR TITLE
Add support for login_hint param

### LINKED ISSUE #142: Add possibility to specify "login_hint" parameter
As I see from the documentation and from the source code there is no way to specify "login_hint" parameter.

Would be nice to have this feature or to provide access to `OpenIdConnectAuthenticationNotifications.RedirectToIdentityProvider` notification.

### PR DESCRIPTION
**Pull request by laura-rodriguez**
* Add support for login_hint
* Refactor existing code to make it easier to add support for other Okta params in the future
* Update Readme with usage examples
* Fix old StyleCop issues.

## Issue \#

Fix #142 
Fix OKTA-326154

## Code

- [x] Unit test(s)
- [x] Implementation

### CODE REVIEW

File: Okta.AspNetCore.Mvc.IntegrationTest/OktaMiddlewareShould.cs [hunk 1]
```diff
@@ -56,6 +56,19 @@ public async Task IncludeIdpInAuthorizeUrlAsync()
             }
         }
 
+        [Fact]
+        public async Task IncludeLoginHintInAuthorizeUrlAsync()
+        {
+            var loginWithIdpEndpoint = string.Format("{0}/Account/LoginWithLoginHint", BaseUrl);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, loginWithIdpEndpoint);
```
Comment (andriizhegurov-okta) [inline Okta.AspNetCore.Mvc.IntegrationTest/OktaMiddlewareShould.cs:63]: Looks like this is unnecessary line: the `request` variable is not used anywhere.

Comment (laura-rodriguez) [reply Okta.AspNetCore.Mvc.IntegrationTest/OktaMiddlewareShould.cs:63]: Good catch! I'll fix it. 👍

Comment (laura-rodriguez) [reply Okta.AspNetCore.Mvc.IntegrationTest/OktaMiddlewareShould.cs]: Fixed.

**Review by andriizhegurov-okta — APPROVED** [review]
:+1:

**Review by bryanapellanes-okta — APPROVED** [review]
👍

**Pull request merged**